tvb_epilepsy/base/equations.py

```diff
@@ -258,8 +258,8 @@ def eqtn_jac_fz_2d(x1, z, tau1, tau0, zmode=array("lin"), z_pos=True, K=None, w=
 
     # Assuming that wii = 0
     jac_x1 += multiply(K, sum(w, 1))
-    jac_x1 = diag(jac_x1.flatten()) - multiply(repeat(reshape(K, (x1.size, 1)), 3, axis=1), w)
-    jac_x1 *= repeat(reshape(tau, (x1.size, 1)), 3, axis=1)
+    jac_x1 = diag(jac_x1.flatten()) - multiply(repeat(reshape(K, (x1.size, 1)), x1.size, axis=1), w)
```

Comment:
These `repeat`s and `reshape`s would be far cleaner if they were written as regular arithmetic and broadcasting operations.